account_payment_order/models/account_move.py

```diff
@@ -57,6 +57,9 @@ def _get_payment_order_communication_direct(self):
                 communication = self.ref or self.payment_reference
             else:
                 communication = self.payment_reference or self.name
+        default_payment_type = self._context.get("default_payment_type", False)
+        if default_payment_type == "inbound":
+            communication = self.name
```

Comment:
Should the name also de assigned if the account.move is not an invoice?

Reply:
Changed so that this only applies when it's an invoice